Removing "allowed buckets" functionality

BarakSason opened the pull request:
The "allowed buckets" functionality interferes with
another access control mechanism which was implemented
and thus needs to be removed

Signed-off-by: Barak Sason Rofman <sason922@gmail.com>

nimrod-becker commented on .settings/org.eclipse.wst.jsdt.ui.superType.container:
```diff
@@ -0,0 +1 @@
+org.eclipse.wst.jsdt.launching.JRE_CONTAINER
```
please remove

nimrod-becker commented on .settings/org.eclipse.wst.jsdt.ui.superType.name:
```diff
@@ -0,0 +1 @@
+Global
```
please remove

nimrod-becker commented:
Notice deepscan new issues and lint issues failing all the tests

jackyalbo commented:
@BarakSason Thinking maybe we should split this PR? maybe one PR for UI, that we could easily push if it doesn't break. and the rest on a second one..  what do you think? will be easier to go through...

jackyalbo commented on src/server/common_services/auth_server.js:
```diff
@@ -591,10 +591,6 @@ function _prepare_auth_request(req) {
         dbg.log3('load auth system:', req.system && req.system._id);
     };
 
-    req.has_bucket_permission = function(bucket) {
```
has_s3_bucket_permission uses this function - not sure you can just remove. Also not sure what is our plan about this once allowed buckets are removed? to deny all requests or to allow them.

jackyalbo commented on src/server/common_services/auth_server.js:
```diff
@@ -591,10 +591,6 @@ function _prepare_auth_request(req) {
         dbg.log3('load auth system:', req.system && req.system._id);
     };
 
-    req.has_bucket_permission = function(bucket) {
```
has_s3_bucket_permission uses this function - not sure you can just remove. Also not sure what is our plan about this once allowed buckets are removed? to deny all requests or to allow them.

BarakSason replied:
@jackyalbo I'm fine with splitting this. @nimrod-becker Your thoughts?
As far as I understand, there is a different mechanism which is used instead of the "allowed buckets" one, so it's not a matter of allowing/denying everything as the other mechanism will take care of it (Nimrod - If I'm mistaken, please correct me).

nimrod-becker replied:
Regarding splitting, we can do that that is ok.

Allow/Deny will be be set according to the bucket policy, I believe, but might be mistaken, that Jacky is asking what would be the default in case there is no explicit rule. We should behave as S3 would, can easily test it on an S3 bucket

guymguym replied:
Hey, I think that the default when there is no bucket policy should be to allow only the owner of the bucket, like AWS does ->

https://docs.aws.amazon.com/AmazonS3/latest/userguide/access-control-overview.html#about-resource-owner

>Amazon S3 bucket and object ownership
>
> Buckets and objects are Amazon S3 resources. By default, only the resource owner can access these resources. The resource owner refers to the AWS account that creates the resource. For example:
> 
> 1. The AWS account that you use to create buckets and upload objects owns those resources.
> 
> 2. If you upload an object using AWS Identity and Access Management (IAM) user or role credentials, the AWS account that the user or role belongs to owns the object.
> 
> 3. A bucket owner can grant cross-account permissions to another AWS account (or users in another account) to upload objects. In this case, the AWS account that uploads objects owns those objects. The bucket owner does not have permissions on the objects that other accounts own, with the following exceptions:
> 
>    - The bucket owner pays the bills. The bucket owner can deny access to any objects, or delete any objects in the bucket, regardless of who owns them.
> 
>     - The bucket owner can archive any objects or restore archived objects regardless of who owns them. Archival refers to the storage class used to store the objects. For more information, see Managing your storage lifecycle.
>

BarakSason committed:
Removing "allowed buckets" functionality

The "allowed buckets" functionality interferes with
another access control mechanism which was implemented
and thus needs to be removed

Signed-off-by: Barak Sason Rofman <sason922@gmail.com>

liranmauda reviewed (commented):
@BarakSason 
Changing the UI requires building it. as I don't see `lib.js` and `lib.js.map` files I assume that it was missed.
Please run:
`cd frontend`
`npm run build-lib`
and then commit those changes (I assume that there will be changes).

liranmauda commented:
> @BarakSason Thinking maybe we should split this PR? maybe one PR for UI, that we could easily push if it doesn't break. and the rest on a second one.. what do you think? will be easier to go through...

@jackyalbo Not sure how it can be split. 
this PR is removing `allowed_buckets` from the schema which will probably break the UI.
WDYT?

jackyalbo commented on src/server/system_services/bucket_server.js:
```diff
@@ -1551,12 +1481,6 @@ function get_bucket_info({
-            func_config.version === trigger.func_version
-        );
-        const exec_account = system_store.get_account_by_email(func.exec_account);
-        ret_trigger.permission_problem = !auth_server.has_bucket_permission(bucket, exec_account);
```
Again - none bucket owner should lead to permission problem

jackyalbo commented:
@BarakSason also I think you are missing on upgrade script - did you test an upgrade? Not sure what you were planning to do if a system already have account with allowed bucket - but it needs to be handled somehow.

nimrod-becker commented:
Closing will update when a new PR replacing this one will come up
